### ISSUE TITLE
[Bug]: builder-webpack5 `Can't resolve 'storybook/internal/docs-tools'` after `8.2.0`

### ISSUE THREAD

**Issue by tony-scio** [storybookjs/storybook#28591]
### Describe the bug

When attempting to upgrade from `8.1.11` to `8.2.0` (or `8.2.2`), I get the error below. It seems like a dependency might need to be added back after the dependency slimming (which was totally awesome btw).

```
SB_BUILDER-WEBPACK5_0002 (WebpackInvocationError): Module not found: Error: Can't resolve 'storybook/internal/docs-tools' in './node_modules/.aspect_rules_js/@storybook+react@8.2.2_-2113478954/node_modules/@storybook/react/dist'
    at ./node_modules/.aspect_rules_js/@storybook+builder-webpack5@8.2.2_303820626/node_modules/@storybook/builder-webpack5/dist/index.js:1:8835
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/HookWebpackError.js:68:3
    at Hook.eval [as callAsync] (eval at create (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/Hook.js:18:14)
    at Cache.shutdown (./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Cache.js:156:23)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compiler.js:1378:15
    at Hook.eval [as callAsync] (eval at create (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/Hook.js:18:14)
    at Compiler.close (./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compiler.js:1371:23)
    at ./node_modules/.aspect_rules_js/@storybook+builder-webpack5@8.2.2_303820626/node_modules/@storybook/builder-webpack5/dist/index.js:1:8820
    at finalCallback (./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compiler.js:496:32)
    at onCompiled (./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compiler.js:510:20)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compiler.js:1326:21
    at eval (eval at create (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/EntryPlugin.js:49:5
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:2279:13
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:2168:6
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:2030:13
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/util/AsyncQueue.js:354:5
    at Hook.eval [as callAsync] (eval at create (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncQueue._handleResult (./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/util/AsyncQueue.js:324:21)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/util/AsyncQueue.js:307:11
    at onTransitiveTasksFinished (./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:1544:20)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:1530:7
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:2030:13
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/util/AsyncQueue.js:354:5
    at Hook.eval [as callAsync] (eval at create (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncQueue._handleResult (./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/util/AsyncQueue.js:324:21)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/util/AsyncQueue.js:307:11
    at onTransitiveTasksFinished (./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:1544:20)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:1530:7
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:1860:14
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/util/AsyncQueue.js:354:5
    at Hook.eval [as callAsync] (eval at create (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncQueue._handleResult (./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/util/AsyncQueue.js:324:21)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/util/AsyncQueue.js:307:11
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/Compilation.js:2105:13
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/NormalModuleFactory.js:904:13
    at eval (eval at create (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/NormalModuleFactory.js:341:22
    at eval (eval at create (./node_modules/.aspect_rules_js/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/NormalModuleFactory.js:518:22
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/NormalModuleFactory.js:151:11
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/NormalModuleFactory.js:776:25
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/NormalModuleFactory.js:988:8
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/NormalModuleFactory.js:1118:5
    at ./node_modules/.aspect_rules_js/neo-async@2.6.2/node_modules/neo-async/async.js:6883:13
    at ./node_modules/.aspect_rules_js/webpack@5.93.0_275684263/node_modules/webpack/lib/NormalModuleFactory.js:1101:45
    at finishWithoutResolve (./node_modules/.aspect_rules_js/enhanced-resolve@5.17.0/node_modules/enhanced-resolve/lib/Resolver.js:567:11)
    at ./node_modules/.aspect_rules_js/enhanced-resolve@5.17.0/node_modules/enhanced-resolve/lib/Resolver.js:656:15

```

### Reproduction link

.

### System

```bash
.
```

**Comment by ndelangen** [comment]
Please supply a reproduction, and I can take a look.

**Comment by kasperpeulen** [comment]
@tony-scio Do you have `storybook` as a dependency in your `package.json`?

Could you share your full `package.json`?

**Comment by ndelangen** [comment]
Did you upgrade using the upgrade command in the storybook CLI?

see: https://storybook.js.org/docs/configure/upgrading#upgrade-script

**Comment by tony-scio** [comment]
Thanks for the responses. I can't share the full package.json, but the relevant portion is:

```diff
--- a/package.json
+++ b/package.json
@@ -197,17 +197,17 @@
     "@playwright/browser-chromium": "1.45.0",
     "@pmmmwh/react-refresh-webpack-plugin": "^0.5.15",
     "@storybook/addon-coverage": "^1.0.4",
-    "@storybook/addon-docs": "8.1.11",
-    "@storybook/addon-essentials": "8.1.11",
-    "@storybook/addon-interactions": "8.1.11",
-    "@storybook/addon-viewport": "8.1.11",
+    "@storybook/addon-docs": "^8.2.4",
+    "@storybook/addon-essentials": "^8.2.4",
+    "@storybook/addon-interactions": "^8.2.4",
+    "@storybook/addon-viewport": "^8.2.4",
     "@storybook/jest": "^0.2.3",
-    "@storybook/preview-api": "8.1.11",
-    "@storybook/react": "8.1.11",
-    "@storybook/react-webpack5": "8.1.11",
-    "@storybook/test": "8.1.11",
+    "@storybook/preview-api": "^8.2.4",
+    "@storybook/react": "^8.2.4",
+    "@storybook/react-webpack5": "^8.2.4",
+    "@storybook/test": "^8.2.4",
     "@storybook/test-runner": "^0.19.0",
-    "@storybook/types": "8.1.11",
+    "@storybook/types": "^8.2.4",
     "@swc/core": "1.4.17",
     "@swc/helpers": "^0.5.12",
     "@swc/jest": "^0.2.36",
@@ -304,7 +304,7 @@
     "qs": "6.12.1",
     "semver": "^7.7.0",
     "sinon-chrome": "^3.0.1",
-    "storybook": "8.1.11",
+    "storybook": "^8.2.4",
     "storybook-addon-pseudo-states": "^3.1.1",
     "storybook-addon-swc": "^1.1.9",
     "storybook-dark-mode": "^4.0.2",
```

Running `npx storybook@latest upgrade` generated the same diff.

**Comment by johannesgoslar-synth** [comment]
We are also running into a similar issue when going from 8.1 to 8.2 when running tests:
```
    Cannot find module 'storybook/internal/preview-api' from '../../node_modules/@storybook/react/dist/index.js'
```
Happens in a pnpm workspace, all storybook stuff is 8.2.4 when checking with pnpm why and the migrations did run.

**Comment by ndelangen** [comment]
@johannesgoslar-synth @tony-scio based on those diffs, I can't tell what's wrong then.

I'd really need a repro to tell what's wrong.

The only idea I have at the moment is that something in webpack is preventing the export-map to be used/respected.
Could be a webpack config, or an outdated webpack version, is my best guess.

**Comment by johannesgoslar-synth** [comment]
Yeah, I understand the need for a repro, but it sadly isn't a public repo, so I can't share it.
Will check if updating webpack helps somehow.

**Comment by kasperpeulen** [comment]
@johannesgoslar-synth Can you maybe share some more info such as:

 **npx storybook info** - Environment and storybook package versions
 **npx storybook doctor** - Health checks for your project

And maybe share some bits of your package.json/main.ts etc.

**Comment by johannesgoslar-synth** [comment]
No issues with doctor.
Info:
```
Storybook Environment Info:

  System:
    OS: macOS 14.5
    CPU: (11) arm64 Apple M3 Pro
    Shell: 3.7.1 - /opt/homebrew/bin/fish
  Binaries:
    Node: 18.20.3 - ~/.nvm/versions/node/v18.20.3/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v18.20.3/bin/yarn
    npm: 10.7.0 - ~/.nvm/versions/node/v18.20.3/bin/npm
  Browsers:
    Chrome: 126.0.6478.182
    Firefox: 128.0
    Safari: 17.5
  npmPackages:
    @storybook/node-logger: ^8.2.4 => 8.2.4 
    @storybook/preview-api: ^8.2.4 => 8.2.4 
    @storybook/react: ^8.2.4 => 8.2.4 
  ```
  
  Moving to the rspack builder resolves the issue. We are moving things out of webpack anyway, so I probably just gonna fix stuff down that route.

**Comment by vreviglio** [comment]
@tony-scio I was facing a similar issue and the problem was caused because I have define a alias resolve for the webpack configuration, so I had something like:

```
  webpackFinal: (config) => {
    if (config.resolve) {
      config.resolve.alias = {
        storybook: path.resolve(__dirname, './'),
      };

      config.resolve.modules = [
        path.resolve(__dirname, '../src'),
        'node_modules',
      ];

      config.resolve.extensions?.push('.scss');
    }

    return config;
  },
  
  ```
  
  The problem was that some dependencies used this same alias, so when they tried to resolve the import statement this alias was breaking the dependencies.

To fix this I had to renamed the alias that we were using in our project. I've tried to renamed the alias to @storybook, but there are other dependencies that cause the same issue.

Let me know if this help you solve the issue
